src/SpeedLimiter.cc

```diff
@@ -97,105 +44,152 @@ class ignition::math::SpeedLimiterPrivate
 };
 
 //////////////////////////////////////////////////
-SpeedLimiter::SpeedLimiter(bool   _hasVelocityLimits,
-                           bool   _hasAccelerationLimits,
-                           bool   _hasJerkLimits,
-                           double _minVelocity,
-                           double _maxVelocity,
-                           double _minAcceleration,
-                           double _maxAcceleration,
-                           double _minJerk,
-                           double _maxJerk)
-  : dataPtr(std::make_unique<SpeedLimiterPrivate>(_hasVelocityLimits,
-      _hasAccelerationLimits, _hasJerkLimits, _minVelocity, _maxVelocity,
-      _minAcceleration, _maxAcceleration, _minJerk, _maxJerk))
+SpeedLimiter::SpeedLimiter()
+  : dataPtr(std::make_unique<SpeedLimiterPrivate>())
 {
 }
 
 //////////////////////////////////////////////////
 SpeedLimiter::~SpeedLimiter() = default;
 
 //////////////////////////////////////////////////
-double SpeedLimiter::Limit(double &_v, double _v0, double _v1,
+void SpeedLimiter::SetMinVelocity(double _lim)
+{
+  this->dataPtr->minVelocity = _lim;
+}
+
+//////////////////////////////////////////////////
+double SpeedLimiter::MinVelocity() const
+{
+  return this->dataPtr->minVelocity;
+}
+
+//////////////////////////////////////////////////
+void SpeedLimiter::SetMaxVelocity(double _lim)
+{
+  this->dataPtr->maxVelocity = _lim;
+}
+
+//////////////////////////////////////////////////
+double SpeedLimiter::MaxVelocity() const
+{
+  return this->dataPtr->maxVelocity;
+}
+
+//////////////////////////////////////////////////
+void SpeedLimiter::SetMinAcceleration(double _lim)
+{
+  this->dataPtr->minAcceleration = _lim;
+}
+
+//////////////////////////////////////////////////
+double SpeedLimiter::MinAcceleration() const
+{
+  return this->dataPtr->minAcceleration;
+}
+
+//////////////////////////////////////////////////
+void SpeedLimiter::SetMaxAcceleration(double _lim)
+{
+  this->dataPtr->maxAcceleration = _lim;
+}
+
+//////////////////////////////////////////////////
+double SpeedLimiter::MaxAcceleration() const
+{
+  return this->dataPtr->maxAcceleration;
+}
+
+//////////////////////////////////////////////////
+void SpeedLimiter::SetMinJerk(double _lim)
+{
+  this->dataPtr->minJerk = _lim;
+}
+
+//////////////////////////////////////////////////
+double SpeedLimiter::MinJerk() const
+{
+  return this->dataPtr->minJerk;
+}
+
+//////////////////////////////////////////////////
+void SpeedLimiter::SetMaxJerk(double _lim)
+{
+  this->dataPtr->maxJerk = _lim;
+}
+
+//////////////////////////////////////////////////
+double SpeedLimiter::MaxJerk() const
+{
+  return this->dataPtr->maxJerk;
+}
+
+//////////////////////////////////////////////////
+double SpeedLimiter::Limit(double &_vel, double _prevVel, double _prevPrevVel,
     std::chrono::steady_clock::duration _dt) const
 {
-  const double tmp = _v;
+  const double vDesired = _vel;
```

Comment:
I think `velInput` or `velInitial` or `velUnclamped` make more sense than `vDesired`. this class is just a speed limiter, not a controller with desired values. If we do rename this, we should do so for the other `vDesired`, `accDesired`, and `jerkDesired` variables as well